src/sentry/tsdb/snuba.py

```diff
@@ -21,19 +32,72 @@ class SnubaTSDB(BaseTSDB):
-        TSDBModel.users_affected_by_group: ("issue", "tags[sentry:user]"),
-        TSDBModel.users_affected_by_project: ("project_id", "tags[sentry:user]"),
-        TSDBModel.frequent_environments_by_group: ("issue", "environment"),
-        TSDBModel.frequent_releases_by_group: ("issue", "tags[sentry:release]"),
-        TSDBModel.frequent_issues_by_project: ("project_id", "issue"),
+        TSDBModel.project: SnubaModelSettings("project_id", None, snuba.Dataset.Events, None),
+        TSDBModel.group: SnubaModelSettings("issue", None, snuba.Dataset.Events, None),
+        TSDBModel.release: SnubaModelSettings(
+            "tags[sentry:release]", None, snuba.Dataset.Events, None
+        ),
+        TSDBModel.users_affected_by_group: SnubaModelSettings(
+            "issue", "tags[sentry:user]", snuba.Dataset.Events, None
+        ),
+        TSDBModel.users_affected_by_project: SnubaModelSettings(
+            "project_id", "tags[sentry:user]", snuba.Dataset.Events, None
+        ),
+        TSDBModel.frequent_environments_by_group: SnubaModelSettings(
+            "issue", "environment", snuba.Dataset.Events, None
+        ),
+        TSDBModel.frequent_releases_by_group: SnubaModelSettings(
+            "issue", "tags[sentry:release]", snuba.Dataset.Events, None
+        ),
+        TSDBModel.frequent_issues_by_project: SnubaModelSettings(
+            "project_id", "issue", snuba.Dataset.Events, None
+        ),
+    }
+
+    model_columns_being_upgraded = {
+        TSDBModel.organization_total_received: SnubaModelSettings(
+            "org_id",
+            "times_seen",
+            snuba.Dataset.Outcomes,
+            [["outcome", "=", outcomes.Outcome.ACCEPTED]],
+        ),
+        TSDBModel.organization_total_rejected: SnubaModelSettings(
+            "org_id",
+            "times_seen",
+            snuba.Dataset.Outcomes,
+            [["outcome", "=", outcomes.Outcome.RATE_LIMITED]],
+        ),
+        TSDBModel.organization_total_blacklisted: SnubaModelSettings(
+            "org_id",
+            "times_seen",
+            snuba.Dataset.Outcomes,
+            [["outcome", "=", outcomes.Outcome.FILTERED]],
+        ),
+        TSDBModel.project_total_received: SnubaModelSettings(
+            "project_id",
+            "times_seen",
+            snuba.Dataset.Outcomes,
+            [["outcome", "=", outcomes.Outcome.ACCEPTED]],
+        ),
+        TSDBModel.project_total_rejected: SnubaModelSettings(
+            "project_id",
+            "times_seen",
+            snuba.Dataset.Outcomes,
+            [["outcome", "=", outcomes.Outcome.RATE_LIMITED]],
+        ),
+        TSDBModel.project_total_blacklisted: SnubaModelSettings(
+            "project_id",
+            "times_seen",
+            snuba.Dataset.Outcomes,
+            [["outcome", "=", outcomes.Outcome.FILTERED]],
+        ),
     }
 
+    all_model_columns = dict(model_columns.items() + model_columns_being_upgraded.items())
```

Comment:
What was the reason for keeping these in two separate dictionaries ?

Reply:
It's cause [getsentry/getsentry:tsdb.py](https://github.com/getsentry/getsentry/blob/04a36b0e5490d9d610b8385d6898ed11813c8ad7/getsentry/tsdb.py#L70) uses `model_columns` to figure out if the query should go to Snuba. So, I made `model_columns_being_upgraded ` go to redis in the main thread but snuba in the background thread.

And then, `all_model_columns` is used by `get_data` and `get_range` don't need to worry about which the model comes from.

Reply:
ok